File: lab2/src/Magazine.java

/**
 * Магазин — компонент, що зберігає кількість патронів.
 */
public class Magazine {
    private final int capacity;
    private int roundsLoaded;

    /** Конструктор. */
    public Magazine(int capacity) {
        this.capacity = Math.max(0, capacity);
        this.roundsLoaded = 0;
    }

    public boolean loadRound() {
        if (roundsLoaded < capacity) {
            roundsLoaded++;
            return true;
        }
        return false;
    }

    public boolean unloadRound() {
        if (roundsLoaded > 0) {
            roundsLoaded--;
            return true;
        }
        return false;
    }

    public void reload(int rounds) {
        if (rounds < 0) rounds = 0;
        roundsLoaded = Math.min(capacity, rounds);
    }

    public int getCapacity() { return capacity; }
    public int getRoundsLoaded() { return roundsLoaded; }
}

File: lab9/pistol_pkg/magazine.py
class Magazine:
    """Резервуар для води або куль."""

    def __init__(self, capacity=100):
        self.capacity = capacity
        self.rounds_loaded = 0

    def available(self):
        return self.rounds_loaded

    def reload(self, amount):
        self.rounds_loaded = min(self.capacity, amount)

    def unload_round(self):
        if self.rounds_loaded > 0:
            self.rounds_loaded -= 1

    def getRoundsLoaded(self):
        return self.rounds_loaded

    def getCapacity(self):
        return self.capacity
